src/pow_fpga_accel.h

```diff
@@ -0,0 +1,13 @@
+#ifndef POW_FPGA_ACCEL_H_
+#define POW_FPGA_ACCEL_H_
+
+#include <fcntl.h>
+#include <sys/mman.h>
```

Comment:
Only `<stdint.h>` is needed in this header. You can safely exclude other header files for inclusion.

Reply:
If the header only includes `<stdint.h>` and `"trinary.h"`, the build process is fail. 
```
...
src/pow_fpga_accel.c:61:17: warning: implicit declaration of function ‘open’; did you mean ‘popen’? [-Wimplicit-function-declaration]
     devmem_fd = open("/dev/mem", O_RDWR | O_SYNC);
                 ^~~~
                 popen
src/pow_fpga_accel.c:61:34: error: ‘O_RDWR’ undeclared (first use in this function)
     devmem_fd = open("/dev/mem", O_RDWR | O_SYNC);
                                  ^~~~~~
src/pow_fpga_accel.c:61:34: note: each undeclared identifier is reported only once for each function it appears in
src/pow_fpga_accel.c:61:43: error: ‘O_SYNC’ undeclared (first use in this function)
     devmem_fd = open("/dev/mem", O_RDWR | O_SYNC);
                                           ^~~~~~
src/pow_fpga_accel.c:72:22: warning: implicit declaration of function ‘mmap’ [-Wimplicit-function-declaration]
         (uint32_t *) mmap(NULL, HPS_TO_FPGA_SPAN, PROT_READ | PROT_WRITE,
                      ^~~~
src/pow_fpga_accel.c:72:51: error: ‘PROT_READ’ undeclared (first use in this function); did you mean ‘__WNOTHREAD’?
         (uint32_t *) mmap(NULL, HPS_TO_FPGA_SPAN, PROT_READ | PROT_WRITE,
                                                   ^~~~~~~~~
                                                   __WNOTHREAD
src/pow_fpga_accel.c:72:63: error: ‘PROT_WRITE’ undeclared (first use in this function); did you mean ‘PROT_READ’?
         (uint32_t *) mmap(NULL, HPS_TO_FPGA_SPAN, PROT_READ | PROT_WRITE,
                                                               ^~~~~~~~~~
                                                               PROT_READ
src/pow_fpga_accel.c:73:27: error: ‘MAP_SHARED’ undeclared (first use in this function)
                           MAP_SHARED, devmem_fd, HPS_TO_FPGA_BASE);
                           ^~~~~~~~~~
src/pow_fpga_accel.c:76:26: error: ‘MAP_FAILED’ undeclared (first use in this function); did you mean ‘MAP_SHARED’?
     if (fpga_regs_map == MAP_FAILED) {
                          ^~~~~~~~~~
                          MAP_SHARED
src/pow_fpga_accel.c:78:9: warning: implicit declaration of function ‘close’; did you mean ‘pclose’? [-Wimplicit-function-declaration]
         close(devmem_fd);
         ^~~~~
         pclose
src/pow_fpga_accel.c: In function ‘pow_fpga_accel_destroy’:
src/pow_fpga_accel.c:96:14: warning: implicit declaration of function ‘munmap’ [-Wimplicit-function-declaration]
     result = munmap(fpga_regs_map, HPS_TO_FPGA_SPAN);
              ^~~~~~
...
```

Reply:
To minimize number of header files, my method is excluding a header file and building the project. If there is no problem, I exclude the header file. Otherwise, I include the header file.

Reply:
You should include the necessary headers in implementation files (usually C/C++ source ) instead of headers.